File: api/Bets.ts
class BetPostBody {
    constructor(public client: string,
                public amount: number,
                public player: number,
                public matchId: number) {
    }

    public toAPIJson() {
        return {
            client: this.client,
            montant: this.amount,
            joueur: this.player,
            partie: this.matchId,
        }
    }
}

class Bet {
    constructor(public clientId: string, public betOnJ1: number, public betOnJ2: number) {
    }
}

class Earning {
    constructor(public clientId: string, public amount: number) {
    }
}

class BetResult {
    static ACCEPTED = "PARI_ACCEPTED"

    constructor(public tag: string, public betOnJ1: number, public betOnJ2: number, public totalJ1: number, public totalJ2: number) {
    }

    static parse(r: any) {
        return new BetResult(r.tag, r.bet_on_j1, r.bet_on_j2, r.total_j1, r.total_j2)
    }
}

File: android/app/src/main/java/ca/usherbrooke/bonpari/api/Bets.kt
package ca.usherbrooke.bonpari.api

import com.squareup.moshi.Json

data class BetPostBody(@Json(name = "client")  val client: String,
                       @Json(name = "montant") val amount: Float,
                       @Json(name = "joueur") val player: Int,
                       @Json(name = "partie") val matchId: Int)

data class Bet(@Json(name = "client") val clientId: String,
               @Json(name = "bet_on_j1") val betOnJ1: Float,
               @Json(name = "bet_on_j2") val betOnJ2: Float) : java.io.Serializable

data class Earning(@Json(name = "client") val clientId: String,
                   @Json(name = "amount") val amount: Float) : java.io.Serializable

data class BetResult(@Json(name = "tag") private val tag: String,
                     @Json(name = "bet_on_j1") val betOnJ1: Float,
                     @Json(name = "bet_on_j2") val betOnJ2: Float,
                     @Json(name = "total_j1") val totalJ1: Float,
                     @Json(name = "total_j2") val totalJ2: Float) {

    fun wasAccepted() = tag == ACCEPTED
    fun wasRefresh() = tag == REFRESH
    fun isClosed() = tag == CLOSED

    companion object {
        const val CLOSED = "PARIS_CLOSED"
        const val ACCEPTED = "PARI_ACCEPTED"
        const val REFRESH = "REFRESH"
    }
}